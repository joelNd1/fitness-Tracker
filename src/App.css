.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 600px;
  margin: 0 auto;
  padding: 50px;
  background-color: #fff;
  border-radius: 10px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}



input {
  border: 1px solid #ccc;
  width: 200px;
  border-radius: 4px;
  padding: 8px 12px;
margin: 5px;  
}

body, html {
  margin: 0;
  padding: 0;
  font-family: Arial, sans-serif;
  background-color: #f2f2f2;
}


h1 {
  color: #333;
  text-align: center;
}

/* Style the form */
label {
  font-weight: bold;
  margin-bottom: 8px;
}

input[type="text"],
input[type="number"],
input[type="date"] {
  /* width: 90%;
  padding: 8px;
  margin-bottom: 12px;
  border: 1px solid #ccc;
  border-radius: 4px; */
  display: flex;
  border-top: 1px solid #ccc;
  padding: 10px;
  background-color: #f5f5f5;
  justify-content: center; /* Horizontal alignment to center */

  
}

button {
  background-color: #007bff;
  color: #fff;
  border: none;
  border-radius: 4px;
  padding: 10px 20px;
  cursor: pointer;
}

button:hover {
  background-color: #0056b3;
}

/* Style the training list */
h1 + h1 {
  margin-top: 20px;
  font-size: 1.2rem;
  color: #333;
}

/* Style the message when there's no training list */
p {
  text-align: center;
  color: #555;
}

/* Style the training list container */
.training-list-container {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  padding: 10px;
}

/* Style the training list items */
.training-list-item {
  margin-bottom: 10px;
  padding: 10px;
  border: 1px solid #ccc;
  border-radius: 4px;
}
