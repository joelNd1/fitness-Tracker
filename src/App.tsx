import React, { useState, useEffect } from 'react';
import Axios from 'axios';
import './App.css';
import { AddSession, UpdateSessiontype, DeleteSession } from './sessionService';

export type GymSession = {
  _id: any;
  Date: Date;
  DayType: string;
}

export function App() {
  const [trainingList, setTrainingList] = useState([] as GymSession[]);
  const [trainingType, setTrainingType] = useState('');
  const [trainingDate, setTrainingDate] = useState('00/00/0000');

  // Destructure the useToggle return values
  const [viewData, setViewData] = useState(false);
  const [newTrainingType, setNewTrainingType] = useState('');

  useEffect(() => {
    Axios.get('http://localhost:3001/read').then((Response) => {
      console.log(Response);
      setTrainingList(Response.data);
    }).catch((err) => {
      // do something here
      console.log('failed');
    });
  }, []);

  return (
    <div className="App">
      <h1>crud 1 with mern</h1>

      <div className="form-container">
        <label>what were you training:</label>
        <input type="text" onChange={(event) => setTrainingType(event.target.value)} />

        <label>what is the date:</label>
        <input type="date" onChange={(event) => setTrainingDate(event.target.value)} />

        <button onClick={()=>AddSession(trainingDate, trainingType)}>Add Session</button>

        <br />
        <br />

        <button data-testid="toggle-button" onClick={()=>setViewData(!viewData)}>Toggle View</button>
      </div>

      {viewData && (
        <div className="training-list-container">
          <h1 data-testid="training-head">Training Record</h1>

          {trainingList.map((val, key) => (
            <div className="training-list-item" key={key}>
              <h1>{val.DayType}</h1>
              <div className="Update-container">
                <input
                  type="text"
                  placeholder="updated version ..."
                  onChange={(event) => setNewTrainingType(event.target.value)}
                />
                <button onClick={() => UpdateSessiontype(val._id, newTrainingType)}>update</button>
              </div>
              <h1>{val.Date.toLocaleString()}</h1>

              <button onClick={() => DeleteSession(val._id)}>Delete</button>
            </div>
          ))}
        </div>
      )}
    </div>
  );
}

export default App;
