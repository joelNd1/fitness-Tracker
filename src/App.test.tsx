import React, { Component } from 'react';
import { RenderResult, act, fireEvent, render, waitFor } from '@testing-library/react';
import App, { GymSession } from './App';

const axios = require('axios');

jest.mock('axios');

describe('toggle button tests', ()=> {
    let component : RenderResult;
    it('toggle button is on screen and rendered as expected', async ()=>{
        axios.get.mockResolvedValue({
          data: [
            {
                _id: 1,
                DayType: 'Push Session',
                Date: new Date(),
            } as GymSession
          ] as GymSession[],
        });
        act(() => {
            component = render(
                <App/>,
            );
        });
        act(() => {
            fireEvent.click(component.getByTestId('toggle-button'));
        });
        await waitFor(() => expect(component.getByText('Push Session')).toBeTruthy());
        expect(component.baseElement).toMatchSnapshot();
        expect(component.getByTestId('toggle-button')).toBeTruthy();
    });
})
 