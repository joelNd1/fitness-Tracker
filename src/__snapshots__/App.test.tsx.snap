// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle button tests toggle button is on screen and rendered as expected 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <h1>
        crud 1 with mern
      </h1>
      <div
        class="form-container"
      >
        <label>
          what were you training:
        </label>
        <input
          type="text"
        />
        <label>
          what is the date:
        </label>
        <input
          type="date"
        />
        <button>
          Add Session
        </button>
        <br />
        <br />
        <button
          data-testid="toggle-button"
        >
          Toggle View
        </button>
      </div>
      <div
        class="training-list-container"
      >
        <h1
          data-testid="training-head"
        >
          Training Record
        </h1>
        <div
          class="training-list-item"
        >
          <h1>
            Push Session
          </h1>
          <div
            class="Update-container"
          >
            <input
              placeholder="updated version ..."
              type="text"
            />
            <button>
              update
            </button>
          </div>
          <h1>
            5/1/2024, 11:13:09 PM
          </h1>
          <button>
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
